vmalinovskiy: fixed file upload

vmalinovskiy opened the pull request.

vmalinovskiy committed:
vmalinovskiy: fixed file upload

vmalinovskiy committed:
vmalinovskiy: fixed file upload (tests update)

TheFrozenFire commented on src/Swagger/Client/Request.php:
```diff
@@ -25,113 +25,115 @@ public function __construct(
-    
-    protected function getDocument()
+
+    public function getDocument()
```
Do these methods need to be made public?

vmalinovskiy replied:
my fault, was made for testing, will revert back

TheFrozenFire commented on src/Swagger/Client/Request.php:
```diff
@@ -25,113 +25,115 @@ public function __construct(
         $client->setMethod($operationReference->getMethod());
-        if($mediaType = $operationConfig->getMediaType()) {
+        $mediaType = $operationConfig->getMediaType();
+        if($mediaType && $mediaType != 'multipart/form-data') {
```
Should use !== in this instance, to avoid weird behaviour.

vmalinovskiy replied:
agree, already read about difference, will fix

TheFrozenFire commented on src/Swagger/Client/OperationConfig.php:
```diff
@@ -6,138 +6,140 @@
     {
-        $this->formParameters['formname'] = $formname;
-        $this->formParameters['filename'] = $filename;
+        $this->formParameters['pathToFile'] = $pathToFile;
```
Was this an upstream API change? Could you point me to the docs for this?

vmalinovskiy replied:
No, it was'nt change of our api.....but i tested it a lot (old our version) and found that test passed just because i accidentally used name 'file' for uploaded file and it was sent as formname.
If you'll use any other name - request was invalid. So i hardcoded formname and added particular parameter name to use as file name. Renamed first parameter to pathToFile for better readability and be sure that users will put there path to file, not simple name of file.

TheFrozenFire replied:
It looks like the requirement for the form field name to be "file" is specific to CallFire's API. Since this is my generic swagger client, this needs to be more flexible and allow any field name to be specified as before.

vmalinovskiy committed:
vmalinovskiy: fixed file upload (updated after review)

vmalinovskiy committed:
vmalinovskiy: reverted file upload, added a way to set name for file

vmalinovskiy commented:
Justin, reverted file upload to the state was before and added a way to set name for file if present.
Added required swagger file parameters as you mentioned in review.
